src/test/java/camp/nextstep/edu/calculator/StringCalculatorTest.java

```diff
@@ -0,0 +1,79 @@
+package camp.nextstep.edu.calculator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class StringCalculatorTest {
+    @DisplayName(":를 구분자로 문자열을 파싱하여 문자열 숫자의 합을 반환한다")
+    @ParameterizedTest
+    @MethodSource("camp.nextstep.edu.calculator.StringCalculatorTestProvider#defaultStrategy")
+    void calculate(String testString, int expected) {
+        //when
+        StringCalculator calculator = new StringCalculator();
+        long result = calculator.calculateSum(testString);
+        //then
+        assertThat(result ).isEqualTo(expected);
+    }
+
+    @DisplayName("null 또는 빈 문자열일 경우 0 반환")
+    @ParameterizedTest
+    @NullAndEmptySource
+    void calculate_given_null_or_empty_string(String testString){
+        //when
+        StringCalculator calculator = new StringCalculator();
+        long result = calculator.calculateSum(testString);
+        //then
+        assertThat(result).isEqualTo(0);
+    }
+
+    @DisplayName("커스텀 구분자로 문자열을 파싱하여 문자열 숫자의 합을 반환한다")
+    @ParameterizedTest
+    @MethodSource("camp.nextstep.edu.calculator.StringCalculatorTestProvider#customStrategy")
+    void calculate_given_custom_splitter(String testString){
+        //when
+        StringCalculator calculator = new StringCalculator();
+        long result = calculator.calculateSum(testString);
+        //then
+        assertThat(result).isEqualTo(0);
+    }
+    @DisplayName("숫자 이외 문자열 또는 음수 문자열이 주어지면 실패한다")
+    @ParameterizedTest
+    @ValueSource(strings =
+        {"a,2", "!,2","-100,2",
+        "//;\\na;2", "//~\\n!:2", "//~\\n-100,2"})
+    void calculate_given_invalid_number(String testString){
+        //when
+        assertThatIllegalArgumentException().isThrownBy(()->{
+            StringCalculator calculator = new StringCalculator();
+            long result = calculator.calculateSum(testString);
+        });
+    }
+
+}
+class StringCalculatorTestProvider {
+    public static Stream<Arguments> defaultStrategy(){
+        return Stream.of(
+            arguments("1,2", 3),
+            arguments("1:2", 3),
+            arguments("1,2:3", 6),
+            arguments("1,2:3,", 6)
+        );
+    }
+    public static Stream<Arguments> customStrategy(){
+        return Stream.of(
+            arguments("//;\\n1;2;3", 6),
```

Comment:
```suggestion
            arguments("//;\n1;2;3", 6),
```

요구사항과 다른 테스트 입력값으로 보입니다.